tidy3d/components/source.py

```diff
@@ -465,6 +465,29 @@ def _dir_vector(self) -> Tuple[float, float, float]:
         return None
 
 
+class BroadbandSource(Source, ABC):
+    """A source with frequency dependent field distributions."""
+
+    num_freqs: pydantic.NonNegativeInt = pydantic.Field(
+        1,
+        title="Number of Frequency Points",
+        description="Number of points to approximate the frequency dependence of injected field.",
+    )
+
+    @cached_property
+    def frequency_grid(self) -> np.ndarray:
```

Comment:
What is the correct output type signature @tylerflex ? `Array`? `ArrayLike[float, 1]`?

Reply:
`np.ndarray` is fine for now. Because it's an output type annotation, it has almost no impact on things. I will re-do the array types at some later time.